### PR TITLE
Code coverage and error handling for CDM Domain Meta export

### PR DESCRIPTION
**Pull request by alondhe**
Re-defined allReports.rda to include "META" report to ensure code coverage by unit tests. Added try-catch logic to handle if there is no CDM_DOMAIN_META table available.

### CODE REVIEW

File: R/exportToJson.R [hunk 1]
```diff
@@ -20,22 +20,24 @@
...
 #                "OBSERVATION",
 #                "OBSERVATION_PERIOD",
-#                "PERSON", 
+#                "PERSON",
 #                "PROCEDURE",
 #                "VISIT",
 #                "MEASUREMENT",
 #                "META")
-# save(allReports,file="allReports.rda")
+# save(allReports,file="data/allReports.rda")
```
Comment (chrisknoll) [inline R/exportToJson.R]: @alondhe : was there a reason for placing this .rda file in a folder 'data'?  This RDA file is not associated to the export to JSON work (where we put the exported json files under a 'data' folder).  Was there some other reason for this change?  I think this caused the creation of another file (see other files added to the commit) in data/allReports.rda, while I think there's already a file calld allReports.rda elsewhere in the repository....

Comment (alondhe) [inline R/exportToJson.R]: @chrisknoll in the checked in version I forked, the allReports.rds file was in "/data", so I just assumed that's where it was supposed to go. Then I changed the path to allReports in the commented section above.

Comment (chrisknoll) [inline R/exportToJson.R]: Ahhh, I see, It was a code comment, not actually executed, OK, that's clear.  Thanks for fixing that, I think it's been wrong in that comment for a while .

**Comment by chrisknoll** [comment]
Ajit, this PR failed on our test server due to the following error (from the log):

Error in parse(outFile) : 
  /home/travis/build/OHDSI/Achilles/Achilles.Rcheck/00_pkg_src/Achilles/R/exportToJson.R:627:23: unexpected '='
626:   )
627:   if (output$MESSAGES =
                      
Can you verify that this builds on your env and push a fix if necessary?

File: R/exportToJson.R [hunk 2]
```diff
@@ -619,9 +621,18 @@ generateDomainMetaReport <- function(conn, dbms, cdmDatabaseSchema, resultsDatab
                                             vocab_database_schema = vocabDatabaseSchema
   )  
   
-  output$MESSAGES <- querySql(conn,queryDomainMeta)
-  jsonOutput = toJSON(output)
-  write(jsonOutput, file=paste(outputPath, "/domainmeta.json", sep=""))  
+  try(
+    output$MESSAGES <- querySql(conn,queryDomainMeta)
+  )
+  if (output$MESSAGES = NULL)
```
Comment (chrisknoll) [inline R/exportToJson.R]: This line seems to result in an error on the build server.  Is it supposed to use == or is there an isNull operator?

Comment (chrisknoll) [inline R/exportToJson.R]: I've found that there is an invalid() call you can make to look for NA/null/empty value (invalid is true if any of those are the case). see: http://svitsrv25.epfl.ch/R-doc/library/gtools/html/invalid.html

Comment (chrisknoll) [inline R/exportToJson.R]: Hi, sorry, but i realize now that it's an an external library, and I don't htink we want to introduce a new dependency to Achilles. So is.null() is something in the base libary that you could use in this case.

**Pull request merged**